### PR TITLE
Add 4 companion apps for applications/audio/fldigi

### PR DESCRIPTION
**Pull request by dysinger**
###### Motivation for this change

There are many companion apps to 'fldigi' missing from nixpkgs. http://www.w1hkj.com/

This app is not really "audio" but rather is a digital communication platform primarily used for ham radio. The companion apps are external to fldigi but integrate into fldigi in the preferences.

2018-10-02-091644_1920x1080_scrot

Side comment: nixpkgs really needs a "ham radio" section in the package tree (not unlike applications/science). There are a ton of ham radio apps. Debian even has a dedicated distro/flavor for hams. Some of the existing ham apps are in "misc" and some in "audio" right now on nixpkgs. Just for now I put these fldigi apps in "misc" along side WSJTX because they are definitely not 'audio' apps.

This is my first P/R to nixpkgs so please tell me all things I need to do to correctly get this finished. I see a big checklist below but all I've done is create the packages and use them on my nixos machine for a few days. They work on Nixos and aren't complicated packages. I imagine they'd work anywhere that fldigi works.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/fllog/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchurl {
+    url = "mirror://sourceforge/fldigi/${name}.tar.gz";
+    sha256 = "042j1g035338vfbl4i9laai8af8iakavar05xn2m4p7ww6x76zzl";
+  };
+
+  buildInputs = [
+    fltk13
+    libjpeg
+    pkgconfig
```
Comment (Mic92) [inline pkgs/applications/misc/fllog/default.nix]: pkgconfig belongs to `nativeBuildInputs`.

File: pkgs/applications/misc/flmsg/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchurl {
+    url = "mirror://sourceforge/fldigi/${name}.tar.gz";
+    sha256 = "1kdlwhxsw02pas9d0kakkq2713wj1m4q881f6am5aq4x8n01f4xw";
+  };
+
+  buildInputs = [
+    fltk13
+    libjpeg
+    pkgconfig
```
Comment (Mic92) [inline pkgs/applications/misc/flmsg/default.nix]: here too

File: pkgs/applications/misc/flrig/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchurl {
+    url = "mirror://sourceforge/fldigi/${name}.tar.gz";
+    sha256 = "1wr7bb2577gha7y3a8m5w60m4xdv8m0199cj2c6349sgbds373w9";
+  };
+
+  buildInputs = [
+    fltk13
+    libjpeg
+    pkgconfig
```
Comment (Mic92) [inline pkgs/applications/misc/flrig/default.nix]: here too

File: pkgs/applications/misc/flwrap/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  src = fetchurl {
+    url = "mirror://sourceforge/fldigi/${name}.tar.gz";
+    sha256 = "0qqivqkkravcg7j45740xfky2q3k7czqpkj6y364qff424q2pppg";
+  };
+
+  buildInputs = [
+    fltk13
+    libjpeg
+    pkgconfig
```
Comment (Mic92) [inline pkgs/applications/misc/flwrap/default.nix]: Here too.

**Review by Mic92 — COMMENTED** [review]
Please also add `autoreconfHook` to `nativeBuildInputs`. It is not really required but regenerating autotools scripts can save us some trouble it put correct paths into the configure files.

**Comment by Mic92** [comment]
Feel free to add a ham directory if you have better insight the software.

**Comment by dysinger** [comment]
ok I will make these changes tonight. Thanks for the feedback.

**Comment by dysinger** [comment]
Simple changes pushed.

**Comment by dysinger** [comment]
I will make a separate commit with all the ham radio app author's pulled in that moves all the ham apps to their own applications/ham-radio folder

**Pull request merged**